lib/kastlex/offsets_cache.ex

```diff
@@ -76,28 +95,29 @@ defmodule Kastlex.OffsetsCache do
                          end)
     request_fields = [replica_id: -1, topics: topics]
     request = :kpro.make_request(:list_offsets, 0, request_fields)
-    endpoints = [{String.to_charlist(leader.host), leader.port}]
-    do_fun = fn(connection) ->
-      {:ok, response} = :kpro.request_sync(connection, request, 10_000)
-      responses = kpro_rsp(response)[:msg][:responses]
-      Enum.each(responses,
-                fn(tr) ->
-                  Enum.each(tr[:partition_responses],
-                            fn(pr) ->
-                              try do
-                                [offset] = pr[:offsets]
-                                :ets.insert(@table, {{tr[:topic], pr[:partition]}, offset})
+    leader = Kastlex.MetadataCache.get_leader(id)
+    client_id = Kastlex.get_brod_client_id()
+    {:ok, conn} = :brod_client.get_connection(client_id, String.to_charlist(leader.host), leader.port)
+    :ok = :kpro.request_async(conn, request)
+    kpro_req(ref: ref) = request
+    new_map = Map.put(state.kafka_requests, ref, leader.node_id)
+    %{state | :kafka_requests => new_map}
+  end
+
+  defp handle_offsets_response(msg) do
+    responses = msg[:responses]
+    Enum.each(responses,
+              fn(tr) ->
+                Enum.each(tr[:partition_responses],
+                          fn(pr) ->
+                            try do
```

Comment:
This is from old code, but perhaps refine it now:
this `try do` is too brutal, a better approach is to respect the `pr.error_code`.
if `:no_error`, bind `offset` with `[offset] = pr[:offsets]` or `offset = pr.offset` depending on the request version.